### PR TITLE
[amp] add gradient clipping for unit tests

### PR DESCRIPTION
**Pull request by 1SAA**

### CODE REVIEW

File: colossalai/amp/naive_amp/naive_amp.py [hunk 1]
```diff
@@ -40,7 +43,7 @@ def step(self):
         return self.optim.step()
 
     def clip_grad_norm(self, model: nn.Module, max_norm: float):
-        pass
+        raise RuntimeError("Misuse error: you should specify the clip_grad_norm in the amp config")
```
Comment (FrankLeeeee) [inline colossalai/amp/naive_amp/naive_amp.py:46]: Perhaps it is clearer to say "NaiveAMP optimizer has executed gradient clipping during optimizer.step() if you have supplied `clip_grad_norm` in the amp_config, executing the method clip_grad_norm is not allowed."

Comment (1SAA) [reply colossalai/amp/naive_amp/naive_amp.py]: @FrankLeeeee done.

**Review by FrankLeeeee — APPROVED** [review]
